### ISSUE TITLE
[bug] Unable to clear the value of an optional enum field

### ISSUE THREAD

**Issue by SimpleCreations** [firecmsco/firecms#508]
The following bug can be observed when specifying an enum field which is not required. Upon clearing the field with the clear button and saving the entity, a validation message can be seen reporting that the value doesn't match any of the enum IDs. However, since the field is not required, `undefined`/`null` should be accepted.

**Field configuration**

```ts
  {
    name: "My enum",
    dataType: "string",
    enumValues: {
      value1: "Value 1",
      value2: "Value 2",
    },
    validation: { required: false },
    clearable: true,
  }
```

**Expected behavior**

When pressing the X button and saving the entity, the entity should get saved without errors.

**Actual behavior**

Before clearing:
<img width="636" alt="Screenshot 2023-06-16 at 11 57 39" src="https://github.com/firecmsco/firecms/assets/12449725/3de97d6d-719c-4961-9b8c-565bd7550a43">

After clearing and saving:
<img width="645" alt="Screenshot 2023-06-16 at 11 58 21" src="https://github.com/firecmsco/firecms/assets/12449725/95eceae4-3a17-4204-b8c6-7c6c0c27833c">

**Comment by fgatti675** [comment]
Thank you! We will check it

**Comment by fgatti675** [comment]
Fixed in version `v2.0.4` :)
